library/manageiq_alert.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/python
+
+
+DOCUMENTATION = '''
+---
+module: manageiq_alert
+description: The manageiq_alert module supports adding, updating and deleting alert definitions in ManageIQ.
+short_description: management of alerts in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  miq_verify_ssl:
+    description:
+      - whether SSL certificates should be verified for HTTPS requests
+    required: false
+    default: True
+    choices: ['True', 'False']
+  ca_bundle_path:
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+  description:
+    description:
+      - the alert definition description in manageiq. this is the primary key
+        used to match alert definitions, therefor always required
+    required: true
+    default: null
+  entity:
+    description:
+      - the entity to base the alert on in manageiq
+    required: false
+    choices: ['container_node', 'vm', 'host', 'storage', 'cluster', 'ems', 'miq_server', 'middleware_server']
+    default: null
+  expression:
+    description:
+      - the expression to evaluate
+    required: false
+    default: null
+  options:
+    description:
+      - Additional alert options, including the notification type and frequency
+    required: false
+    default: null
+  enabled:
+    description:
+      - whether the alert is active or not
+    required: false
+    default: true
+  state:
+    description:
+      - the state of the alert definition
+      - On present, it will create the alert if it does not exist, or update it
+        if the associated data is different
+      - On absent, it will delete the alert if it exists
+    required: false
+    choices: ['present', 'absent']
+    default: 'present'
+'''
+
+EXAMPLES = '''
+# Create a new alert in ManageIQ
+  manageiq_alert:
+    description: Test Alert 01
+    options:
+      notifications:
+        delay_next_evaluation: 60
+        evm_event: {}
+    entity: container_node
+    expression:
```

Comment:
`expression` can be either miq_expression or  hash https://github.com/moolitayer/manageiq/blob/b3af72f325ff42f2743154ce8c796b6b12084aeb/app/models/miq_alert.rb#L328-L331

The way we pass it now seems to always create a miq_expression instead of a hash.
```
#<MiqExpression:0x000000088d3f30 @exp={":mode"=>"internal", ":eval_method"=>"dwh_generic"}, @context_type=nil>
```

The reason is here:
https://github.com/manageiq/manageiq/blob/ca15956dca10d194cd683106a72c1c8e339258fa/app/controllers/api/alert_definitions_controller.rb#L9

So: we will need a way to allow both types. I'm not sure how to do in the API, talk to those guys.
We might need another field in the module evaltualy